core/foundation/inc/ROOT/StringUtils.hxx

```diff
@@ -23,6 +23,25 @@ namespace ROOT {
 
 std::vector<std::string> Split(std::string_view str, std::string_view delims, bool skipEmpty = false);
 
+std::string Join(const std::string& sep, const std::vector<std::string>& strings);
```

Comment:
```suggestion
template <class StringCollection_t>
std::string Join(const std::string &sep, StringCollection_t &&strings)
{
   if (strings.empty())
      return "";
   return std::accumulate(std::next(std::begin(strings)), std::end(strings), strings[0],
                          [&sep](auto const &a, auto const &b) { return a + sep + b; });
}
```

Why restrict ourselves to `std::vector<std::string>`? This has several disadvantages:
1. Memory allocation on the heap always necessary because of the `std::vector<>`. We also should deal with `std::span<>`, `std::initialize_list<>`, etc.
2. If you want to concatenate C-style strings, they first have to be copied to a `std::string`

May I suggest the following templated approach?

Reply:
Also, don't forget to use the `.clang-format` style from the ROOT repo